### PR TITLE
Fix owner reference for thin runtime

### PR DESCRIPTION
**Pull request by cheyang**
Signed-off-by: cheyang <cheyang@163.com>

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/utils/dataset_runtime.go [hunk 1]
```diff
@@ -40,12 +43,24 @@ func GetRuntimeByCategory(runtimes []datav1alpha1.Runtime, category common.Categ
...
+			runtimeToUpdate := runtime.DeepCopy()
+			runtimeToUpdate.SetOwnerReferences([]metav1.OwnerReference{
+				{
+					Kind:       dataset.GetObjectKind().GroupVersionKind().Kind,
+					APIVersion: dataset.APIVersion,
+					Name:       dataset.GetName(),
+					UID:        dataset.GetUID(),
+					Controller: utilpointer.BoolPtr(true),
+				}})
+			if !reflect.DeepEqual(runtimeToUpdate, runtime) {
```
Comment (TrafalgarZZZ) [inline pkg/utils/dataset_runtime.go:60]: I think the code may attempt to create runtime for multiple times: In cases that `runtimeToUpdate` is deep equal to `runtime`, the function will not return and go into the next `if IgnoreNotFound(err) == nil` statement.

File: pkg/utils/dataset_runtime.go [hunk 2]
```diff
@@ -40,12 +43,24 @@ func GetRuntimeByCategory(runtimes []datav1alpha1.Runtime, category common.Categ
...
+				err = client.Update(context.TODO(), runtimeToUpdate)
+				return err
+			}
```
Comment (TrafalgarZZZ) [inline pkg/utils/dataset_runtime.go:63]: Suggest to `return nil` in the end of the `if err == nil` statement for the reason mentioned above.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: pkg/utils/dataset_runtime.go (hunk 1 above)
Comment (cheyang) [reply pkg/utils/dataset_runtime.go:60]: Thanks, I've added `return` in this `if` block.

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

**Comment by xliuqq** [comment]
Adding owner references at creation time is unnecessary? runtime controller reconcile will add the own reference for the auto-created thin runtime

**Pull request merged**